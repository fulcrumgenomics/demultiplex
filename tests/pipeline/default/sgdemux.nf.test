nextflow_pipeline {

    name "Test SGDEMUX"
    script "main.nf"

    test("Should run without failures") {

        when {
            params {
                input = "$baseDir/assets/inputs/sgdemux-samplesheet.csv"
                demultiplexer = 'sgdemux'
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.succeeded().size() == 104

            assert snapshot(
                path("$outputDir/sim-data/metrics.tsv"),
                path("$outputDir/sim-data/per_project_metrics.tsv"),
                path("$outputDir/sim-data/per_sample_metrics.tsv"),
                path("$outputDir/sim-data/sample_barcode_hop_metrics.tsv")
            ).match()
        }

    }

}
