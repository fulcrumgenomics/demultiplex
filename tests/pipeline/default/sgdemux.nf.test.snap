{
    "Should run without failures": {
        "content": [
        "metrics.tsv:md5,7627e4bd2a56ea551fd74b4a2b5cb4b2",
        "per_project_metrics.tsv:md5,7393bc580c8ed8431107ad3dd00a7d2a",
        "per_sample_metrics.tsv:md5,fb8ce024ad2508e7b22a4af65fe44358",
        "sample_barcode_hop_metrics.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
        ],
        "timestamp": "2023-01-18T23:52:50+0000"
    }
}